cmd/gitserver/server/server.go

```diff
@@ -333,11 +333,15 @@ func (s *Server) handleIsRepoCloneable(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	req.Repo = protocol.NormalizeRepo(req.Repo)
 
 	if req.URL == "" {
+		if req.Repo == "" {
+			http.Error(w, "no URL or Repo", http.StatusBadRequest)
```

Comment:
nit: 
```suggestion
			http.Error(w, "no URL and Repo", http.StatusBadRequest)
```

The logic seems to have both ` == ""`.